### PR TITLE
Add fix for older connection plugins

### PR DESCRIPTION
**Pull request by jbemmel**
Resubmission of https://github.com/ansible/ansible/pull/79372

### CODE REVIEW

File: test/support/network-integration/collections/ansible_collections/ansible/netcommon/plugins/connection/older_plugins.py [hunk 1]
```diff
@@ -0,0 +1,1391 @@
...
+        return func(self, *args, **kwargs)
+
+    return wrapped
+
+
+class AnsibleCmdRespRecv(Exception):
+    pass
+
+
+class OldConnection(NetworkConnectionBase):
```
Comment (s-hertel) [inline test/support/network-integration/collections/ansible_collections/ansible/netcommon/plugins/connection/older_plugins.py]: The plugin loader looks for a class called `Connection`.

If you're still considering how to test this, I got the unit test working locally with `--docker` in test/units/executor/test_task_executor.py like this:

```python
from ansible.inventory.host import Host  # FIXME: move new imports to top
import ansible.plugins.loader as loader
from ansible.playbook import Playbook
from ansible.playbook.play_context import PlayContext
import pytest


def test_legacy_networking_connection(tmpdir, mocker):
    tmpdir.mkdir("connection_plugins").join("legacy_connection.py").write("""
from ansible.plugins.connection import NetworkConnectionBase

class Connection(NetworkConnectionBase):
    def _connect(self):
        pass

    @property
    def transport(self):
        pass

    def get_option_and_origin(self, option, hostvars=None):
        if option == 'persistent_log_messages':
            return ('fromini', False)
        elif option == 'persistent_command_timeout':
            return ('fromini', 10)
        return super().get_option_and_origin(option, hostvars)
""")

    pb = """
- hosts: all
  connection: legacy_connection
  gather_facts: false
  tasks:
  - ping:
"""

    loader.init_plugin_loader()
    loader.add_all_plugin_dirs(tmpdir)

    mock_data_loader = DictDataLoader({'playbook.yml': pb})

    mock_var_manager = MagicMock()
    mock_var_manager.get_delegated_vars_and_hostname.return_value = {}, None

    playbook_ds = Playbook.load('playbook.yml', loader=mock_data_loader, variable_manager=mock_var_manager)
    play = playbook_ds._entries[0]

    te = TaskExecutor(
        host=Host('dummy'),
        task=play.tasks[0].block[0],
        job_vars={},
        play_context=PlayContext(play=play),
        new_stdin=None,
        loader=mock_data_loader,
        shared_loader_obj=loader,
        final_q=None,
        variable_manager=mock_var_manager
    )

    # test loading the connection plugin but short-circuit actually establishing a connection
    mocker.patch('ansible.executor.task_executor.start_connection', MagicMock(side_effect=NotImplementedError('success')))
    with pytest.raises(NotImplementedError, match='success'):
        te._execute()
```

**Comment by slefol** [comment]
I don't understand why this issue is stuck.
can you provide additional information please ?

**Comment by bcoca** [comment]
@slefol at least one reason, is that tests are not passing

**Comment by slefol** [comment]
Hello.
Can someone move this forward please?
Me, I don’t know what to do.
This issue is preventing me from upgrading to an Ansible version higher than 2.12.8.
Thanks in advance.

**Comment by s-hertel** [comment]
@slefol @jbemmel I've opened a PR with the patch from this PR with a test to prevent regression https://github.com/ansible/ansible/pull/82954.

**Pull request closed without merging**